sunpy/map/tests/test_mapbase.py

```diff
@@ -1028,31 +1028,24 @@ def test_rotate(aia171_test_map):
     assert aia171_test_map_crop_rot.data.shape[0] < aia171_test_map_crop_rot.data.shape[1]
 
 
-@pytest.mark.xfail(version.parse(np.__version__) >= version.parse("1.20.0"),
-                   reason="Numpy >= 1.20.0 doesn't allow NaN to int conversion")
-def test_rotate_with_incompatible_missing_dtype_warning():
+def test_rotate_with_incompatible_missing_dtype():
     data = np.arange(0, 100).reshape(10, 10)
     coord = SkyCoord(0 * u.arcsec, 0 * u.arcsec, obstime='2013-10-28',
                      observer='earth', frame=sunpy.coordinates.Helioprojective)
     header = sunpy.map.make_fitswcs_header(data, coord)
     test_map = sunpy.map.Map(data, header)
-    with pytest.warns(SunpyUserWarning,
-                      match="The specified `missing` value is not an integer, but the data "
-                      "array is of integer type, so the output may be strange."):
-        test_map.rotate(order=3, missing=np.nan)
-
-
-@pytest.mark.skipif(version.parse(np.__version__) <= version.parse("1.20.0"),
-                    reason="Numpy >= 1.20.0 doesn't allow NaN to int conversion")
-def test_rotate_with_incompatible_missing_dtype_error():
-    data = np.arange(0, 100).reshape(10, 10)
-    coord = SkyCoord(0 * u.arcsec, 0 * u.arcsec, obstime='2013-10-28',
-                     observer='earth', frame=sunpy.coordinates.Helioprojective)
-    header = sunpy.map.make_fitswcs_header(data, coord)
-    test_map = sunpy.map.Map(data, header)
-    with pytest.raises(ValueError, match="cannot convert float NaN to integer"):
-        test_map.rotate(order=3, missing=np.nan)
 
+    if version.parse(np.__version__) >= version.parse("1.20.0"):
```

Comment:
Can you leave these tests as separate unit tests?

This version check is also not needed.

Reply:
But the version check is making sure that we are giving warning to under 1.20.0 numpy user, as any version equal to or above it will fail in it so will provide error rather than a warning? without it or xfail pytest will fail

Reply:
The error should be provided to the user. We also do not support any numpy version below 1.21.0.

Reply:
But do we even need the warning test then? Shouldn't we just have one test for error maybe?
And even in the main code, shouldn't we just omit the if condition?
`
           if version.parse(np.__version__) < version.parse("1.20.0") and issubclass(self.data.dtype.type, numbers.Integral) and (missing % 1 != 0):
            warn_user("The specified `missing` value is not an integer, but the data "
                      "array is of integer type, so the output may be strange.")
`

Reply:
That sounds like a good idea.